compiler/rustc_target/src/spec/targets/wasm32v1_none.rs

```diff
@@ -0,0 +1,88 @@
+//! A "bare wasm" target representing a WebAssembly output that makes zero
+//! assumptions about its environment, similar to wasm32-unknown-unknown, but
+//! that also specifies an _upper_ bound on the set of wasm proposals that are
+//! supported.
+//!
+//! It is implemented as a variant on LLVM's wasm32-unknown-unknown target, with
+//! the additional flags `-Ctarget-cpu=mvp` and `-Ctarget-feature=+mutable-globals`.
+//!
+//! This target exists to resolve a tension in Rustc's choice of WebAssembly
+//! proposals to support. Since most WebAssembly users are in fact _on the web_
+//! and web browsers are frequently updated with support for the latest
+//! features, it is reasonable for Rustc to generate wasm code that exploits new
+//! WebAssembly proposals as they gain browser support. At least by default. And
+//! this is what the wasm32-unknown-unknown target does, which means that the
+//! _exact_ WebAssembly features that Rustc generates will change over time.
+//!
+//! But a different set of users -- smaller but nonetheless worth supporting --
+//! are using WebAssembly in implementations that either don't get updated very
+//! often, or need to prioritize stability, implementation simplicity or
+//! security over feature support. This target is for them, and it promises that
+//! the wasm code it generates will not go beyond the proposals/features of the
+//! W3C WebAssembly core 1.0 spec, which (as far as I can tell) is approximately
+//! "the wasm MVP plus mutable globals". Mutable globals was proposed in 2018
+//! and made it in.
+//!
+//! See https://www.w3.org/TR/wasm-core-1/
+//!
+//! Notably this feature-set _excludes_:
+//!
+//!   - sign-extension operators
+//!   - non-trapping / saturating float-to-int conversions
+//!   - multi-value
+//!   - reference types
+//!   - bulk memory operations
+//!   - SIMD
```

Comment:
Suggestion: use the "official" wasm proposal names here, like `reference-types`? Also below.

Reply:
Done.